### DISCUSSION TITLE
Failed to identify devices connected via SmartThings integration

### DISCUSSION THREAD

**Discussion by tonyspence** [andrew-codechimp/HA-Battery-Notes#644] [Q&A] [answered]
Installed Battery Notes yesterday and impressed with what it provides. Identified battery devices connected through ZHA and via the Wiser integration however it failed to identify devices (zigbee door sensors and motion sensors) connected via the SmartThings integration. Before I attempt to add these 'missing' devices manually (or move then from SmartThings to ZHA) is there anything I can do to enable auto discovery?

**Comment by andrew-codechimp** [comment]
Hi, the library doesn't have all devices unfortunately but it is growing.

You can manually add the device straight away by doing the following
https://andrew-codechimp.github.io/HA-Battery-Notes/#to-add-a-battery-note-manually

You can also contribute missing devices to the library by using this form (it'll take a day or so for it to appear)
https://github.com/andrew-codechimp/HA-Battery-Notes/issues/new?template=new_device_request.yml&title=[Device]%3A+

**Reply by tonyspence** [reply, marked as answer]
I suspect that it's more to do with the SmartThings integration than lack of data in your database. Looking at the devices in HA the manufacturer and model are not populated (other than Samsung products) which is going to make identification a tad difficult. Seems like my best option will be to remove the devices from SmartThings and connect to ZHA.

**Reply by andrew-codechimp** [reply]
Interesting that SmartThings doesn't present the model, I do see SmartThings in the library but I guess they must be via ZHA.

https://github.com/andrew-codechimp/HA-Battery-Notes/blob/main/library.md

**Reply by tonyspence** [reply]
To be fair I've not looked at the SmartThings data for quite some time but as I recall it used to populate these values (I could be wrong). Whether it's a SmartThings change or something in HA which has caused the loss it's probably not something which will be fixed as fast as I can move devices and update a handful of automations. I was always planning to move away from ST so it looks like this is the push I needed.
